JAICore/jaicore-components/src/main/java/ai/libs/jaicore/components/model/ComponentUtil.java

```diff
@@ -414,11 +414,46 @@ public static IComponentInstance getRandomInstantiationOfComponent(final ICompon
 
 			List<IComponent> pluginComponents = new ArrayList<>(ComponentUtil.getComponentsProvidingInterface(componentRepository, reqI.getName()));
 			List<IComponentInstance> satCIList = new ArrayList<>();
-			for (int i = 0; i < numSatisfactions; i++) {
-				satCIList.add(getRandomInstantiationOfComponent(pluginComponents.get(random.nextInt(pluginComponents.size())), componentRepository, random));
+			while (satCIList.size() < numSatisfactions) {
+				IComponentInstance randomCI = getRandomInstantiationOfComponent(pluginComponents.get(random.nextInt(pluginComponents.size())), componentRepository, random);
+				if (reqI.isUniqueComponents()) {
+					if (satCIList.isEmpty() || !isAlreadyContained(satCIList, randomCI)) {
+						satCIList.add(randomCI);
+					}
+				} else {
+					satCIList.add(randomCI);
+				}
 			}
 			ci.getSatisfactionOfRequiredInterfaces().put(reqI.getId(), satCIList);
 		}
 		return ci;
-  }
+	}
+
+	private static boolean isAlreadyContained(final List<IComponentInstance> satCIList, final IComponentInstance componentInstance) {
+		boolean isAlreadyContained = false;
+		for (IComponentInstance satCI : satCIList) {
+			if (areSame(satCI, componentInstance)) {
+				isAlreadyContained = true;
+			}
+		}
+		return isAlreadyContained;
+	}
+
+	public static boolean areSame(final IComponentInstance c1, final IComponentInstance c2) {
```

Comment:
Uniqueness property requires different components (i.e. different component names) not different parametrizations.